File: lessons/day_06/java/src/main/java/com/example/DemoGraphQL/input/InputSkillCreate.java
package com.example.DemoGraphQL.input;

public class InputSkillCreate {
    private Long parent;
    private String name;

    public InputSkillCreate() {
    }

    public InputSkillCreate(Long parent, String name) {
        this.parent = parent;
        this.name = name;
    }

    public Long getParent() {
        return parent;
    }

    public void setParent(Long parent) {
        this.parent = parent;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: lessons/day_05/netcore/src/GraphQLSample/Models/Input/InputSkillCreate.cs
namespace GraphQLNetCore.Models.Input
{
   public class InputSkillCreate
   {
      public string Name { get; set; }

      public int? Parent { get; set; }

      internal Skill ToSkill()
      {
         return new Skill { 
            Name = Name,
            ParentId = Parent
         };
      }
   }
}
